src/core/server/saved_objects/service/lib/crypto/crypto_cli.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  RawAesKeyringNode,
+  buildClient,
+  CommitmentPolicy,
+  RawAesWrappingSuiteIdentifier,
+} from '@aws-crypto/client-node';
```

Comment:
try avoid to introduce new dependency , use build-in library is first choice to reduce maintenance effort

Reply:
We can talk about it during the design review. However, when compared to node crypto, this open-source SDK provides better security practice. For instance, the encryption key generated with KDF and then encrypted with the wrapping key.

Reply:
Our scenario don't need to generate key in node js run time. Let's not spending time to research, and use  build-in module

Reply:
Research is done on https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1756.

Keeping the original text of the encryption key on disk is not secure, so KDF is recommended. I'll also gather some information from the node community.

Reply:
> seems `aws-crypto` provides keyring which can help to protect the encryption key using wrapper key, and may simplify the integration with key store like KMS. @noCharger can you confirm this? I don't mind to introduce `aws-crypto` as new dependency if we can get those benefits

Yes, that is correct. However, @seraphjiang is opposed to involving external dependencies and we noticed SSH storing the private key on local machine.

I believe it is determined by the level of security desired. I'll propose using node crypto and storing the encryption key locally, and I'll seek advice from AppSec.

Reply:
As previously discussed, keep AWS encryption SDK as a dependency. Will refactor the code aimed at merging.